src/backend/cdb/cdbquerycontextdispatching.c

```diff
@@ -1747,6 +1774,7 @@ prepareDispatchedCatalogExternalTable(QueryContextInfo *cxt,
 	{
 		Insist(array_size == 1);
 		AddFileSystemCredentialForPxfTable(location);
```

Comment:
is this still needed? AddFileSystemCredentialForPxfTable as you dispatch it through query context now.